runtime/fiber.c

```diff
@@ -583,3 +591,14 @@ CAMLprim value caml_drop_continuation (value cont)
   caml_free_stack(stk);
   return Val_unit;
 }
+
+CAMLprim value caml_fiber_id(value unit)
+{
+  struct stack_info* stack = Caml_state->current_stack;
+
+  CAMLassert(unit == Val_unit);
+  if (!stack)
+    return caml_copy_int64(-1);
```

Comment:
what are the circumstances under which there's no fibre stack, and we return -1?  Shouldn't retrieving a stackless fibre result in an exception or an assertion failure?

Reply:
Agree. There will always be a fiber when executing OCaml code. The check can be converted to an assertion.